adyen.php

```diff
@@ -569,25 +570,71 @@ public function hookOrderConfirmation()
     public function hookPaymentOptions()
     {
         $payment_options = array();
+        if(!$this->context->customer->is_guest) {
+            $amount = $this->context->cart->getOrderTotal();
+            $currency = $this->context->currency->iso_code;
+            $address = new Address($this->context->cart->id_address_invoice);
+            $isoAddress = Country::getIsoById($address->id_country);
+            $shopperReference = $this->context->cart->id_customer;
+            $shopperLocale = $this->helper_data->getLocale($this->context);
+
+
+            //retrieve payment methods
+            $paymentMethods = $this->helper_data->fetchPaymentMethods($isoAddress, $amount, $currency, $shopperReference, $shopperLocale);
+
+            if (!empty($paymentMethods['oneClickPaymentMethods'])) {
+                $oneClickPaymentMethods = $paymentMethods['oneClickPaymentMethods'];
+                foreach ($oneClickPaymentMethods as $storedCard) {
+                    if (!empty($storedCard["storedDetails"]["card"]["expiryMonth"])) {
```

Comment:
Why is this checking for expiry month? Can we make the reason more obvious somehow?